### DISCUSSION TITLE
How can I prevent −2x + 3 − 2 from turning into 3−2(x+1)?

### DISCUSSION THREAD

**Discussion by meramos** [josdejong/mathjs#3093] [Q&A]
When I use the `simplify` method of the `mathjs` library to simplify the expression `-2x + 3 - 2`, I expect to get `-2x + 1`, but instead I get `3−2(x+1)`. How can I prevent that factorization?

Here is an example of the code I have:
```
import * as math from 'mathjs';

const expression = "-2x + 3 - 2";
const simplifiedExpression = String(math.simplify(expression));
```

I am also having a similar issue with the expression `2x + 2`, which is being simplified to `2(x+1)`.

**Comment by josdejong** [comment]
Thanks for reporting. Are you using the latest version of mathjs? I tried with the latest version of mathjs, this gives:

```js
console.log(math.simplify('-2x + 3 - 2').toString()) // 1 - 2 * x
console.log(math.simplify('2x + 2').toString())      // 2 * (x + 1)
```

So it looks like the first case works ok but the second doesn't make much sense. To solve this we have to tune the rules of `simplify`.

Anyone able to help out here?

**Reply by gwhitney** [reply]
What's wrong with the simplification 2x + 2 -> 2(x + 1)? It has factored out a common factor, which is something we want simplify to do. I mean, shouldn't `10000x + 10000` simplify to `10000(x+1)`? This is the same, just with a smaller number...
